### PR TITLE
Do not set vlanID when its zero

### PR DESCRIPTION
**Pull request by tssurya**
See https://issues.redhat.com/browse/FDP-38 for details

**- What this PR does and why is it needed**

**- Special notes for reviewers**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

**Review by igsilya — APPROVED** [review]
A quote from ovn-nb.5 man page:
```
       tag_request: optional integer, in range 0 to 4,095
              The  VLAN  tag  in  the  network  traffic associated with a con‐
              tainer’s network interface. The client can request ovn-northd to
              allocate  a  tag  that  is unique within the scope of a specific
              parent (specified in parent_name) by setting a  value  of  0  in
              this column. The allocated value is written by ovn-northd in the
              tag column.
```
ovn-kubernetes doesn't use parent ports, so there is no need to ask ovn-northd to allocate tags.

**Review by trozet — APPROVED** [review]

**Comment by tssurya** [comment]
> pgrades and detecting the vlan tag 0 and unsetting it on the por

good point, we need to fix the upgrade scenarios as well

File: go-controller/pkg/ovn/gateway_init.go [hunk 1]
```diff
@@ -521,7 +521,7 @@ func (oc *DefaultNetworkController) addExternalSwitch(prefix, interfaceID, nodeN
 		},
 		Name: interfaceID,
 	}
-	if vlanID != nil {
```
Comment (tssurya) [inline go-controller/pkg/ovn/gateway_init.go:524]: I was speaking to @jcaamano about the upgrades part and then he rightfully pointed out that we are creating new model for the switch port without the tag request, so existing switch ports's tag requests will get auto updated

so no need for any additional logic. I'll make sure QE tests this either ways and update you back here.

Comment (trozet) [reply go-controller/pkg/ovn/gateway_init.go:524]: tested it and it works

**Pull request merged**